File: backend-springboot/src/main/java/com/project/emailSenderApp/services/EmailService.java
package com.project.emailSenderApp.services;

import java.io.File;
import java.io.InputStream;

public interface EmailService {

//    method to send email to single person
    void sendEmail(String to, String subject, String message);

//    method to send email to multiple persons
    void sendEmail(String [] to, String subject, String message);

//    method to send files in email
    void sendEmailWithFile(String to, String subject, String message, File file);

//    method to send html content with email
    void sendEmailWithHtml(String to, String subject, String htmlContent);

//    method to send files in email
    void sendEmailWithFile(String to, String subject, String message, InputStream inputStream);


}

File: frontend-react/src/services/email.service.js
import { customAxios } from "./helper";


export async function sendEmail(emaildata) {
    const data = (await customAxios.post(`/email/send`, emaildata)).data;
    return data;
}

export async function sendEmailWithFile(emaildata) {
    const data = (await customAxios.post(`/email/send-with-file`, emaildata)).data
    return data;
}